src/main/java/org/cactoos/text/CapitalizedText.java

```diff
@@ -53,15 +53,15 @@ public CapitalizedText(final Text text) {
     @Override
     public String asString() throws IOException {
         return new JoinedText(
-            new TransformedIterable<String, Text>(
+            new TransformedIterable<Text, Text>(
                 new SplitText(this.origin),
-                new Func<String, Text>() {
+                new Func<Text, Text>() {
                     @Override
-                    public Text apply(final String input) throws Exception {
+                    public Text apply(final Text input) throws Exception {
```

Comment:
@Timmeey You're proposing something like that?
```java
public CapitalizedText(final Text text) {
   this(
      text,
      new Func<Text, Text>() {
          @Override
          public Text apply(final Text input) throws Exception {
             return new FormattedText(
               "%s%s",
               input.asString().substring(0, 1).toUpperCase(),
               input.asString().substring(1)
            );
         }         
      };
   );
}
public CapitalizedText(final Text text, final Func<Text, Text> func) {
  this.origin = text;
  this.func = func;
}
```
If yes, I liked this proposition. 😄